### PR TITLE
Fix hanging Resharper/Rider testrunner when tests fail

### PR DESCRIPTION
**Pull request by bart-vmware**
## Description

Supplement changes from #958 with try/finally, to avoid hanging Resharper/Rider testrunner when tests fail.

## Quality checklist

- [x] Your code complies with our [Coding Style](https://github.com/SteeltoeOSS/.github/blob/main/contributing-docs/contributing-code-style.md).
- [x] You've updated unit and/or integration tests for your change, where applicable.
- [ ] You've updated documentation for your change, where applicable.
      If your change affects other repositories, such as [Documentation](https://github.com/SteeltoeOSS/Documentation), [Samples](https://github.com/SteeltoeOSS/Samples) and/or [MainSite](https://github.com/SteeltoeOSS/MainSite), add linked PRs here.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.
- [ ] You've added required license files and/or file headers (explaining where the code came from with proper attribution), where code is copied from StackOverflow, a blog, or OSS.

### CODE REVIEW

**Review by TimHess — DISMISSED** [review]

File: src/Stream/test/BinderRabbitMQ.Test/RabbitBinderTests.cs [hunk 1]
```diff
@@ -1516,133 +1531,144 @@ public void TestInternalHeadersNotPropagated()
     [Trait("Category", "SkipOnLinux")]
```
Comment (TimHess) [inline src/Stream/test/BinderRabbitMQ.Test/RabbitBinderTests.cs:1531]: Hm... this test isn't running at all in the CI environment

Comment (bart-vmware) [reply src/Stream/test/BinderRabbitMQ.Test/RabbitBinderTests.cs:1531]: It succeeds locally. Do I need to change something?

Comment (TimHess) [reply src/Stream/test/BinderRabbitMQ.Test/RabbitBinderTests.cs:1531]: I don't know why it is skipped on linux... we should try removing the attribute and seeing what happens, but that does not have to be in this PR

Comment (bart-vmware) [reply src/Stream/test/BinderRabbitMQ.Test/RabbitBinderTests.cs:1531]: Tracked at #995.

**Review by TimHess — APPROVED** [review]

**Pull request merged**